lgn-worker/src/main.rs

```diff
@@ -735,16 +748,6 @@ fn run_with_websocket(config: &Config) -> Result<()>
         .public_params
         .skip_checksum
     {
-        let checksum_url = &config
-            .public_params
-            .checksum_url;
-        let expected_checksums_file = &config
-            .public_params
-            .checksum_expected_local_path;
-        fetch_checksum_file(
-            checksum_url,
-            expected_checksums_file,
-        )?;
```

Comment:
I had moved the download of the checksum file inside this condition, that only runs if the checksum flag is set to `true`. I assumed the file was only used in this branch, the problem is that this assumption was incorrect, the prove constructors also check the checksum, but they don't download the file, so the download must be done before creating the provers.